### PR TITLE
skip miopen batchnorm if channels_last (NHWC)

### PR DESCRIPTION
**Pull request by micmelesse**
MIOPEN does not support channel last tensors so we use the native pytorch implementation.

https://ontrack-internal.amd.com/browse/SWDEV-333420

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
@micmelesse Can you please paste the output of running the two PyTorch unit tests that test Batchnorm with NHWC?

**Comment by micmelesse** [comment]
```
/dockerx/pytorch/test
test_memory_format_nn_BatchNorm2d_cuda_float32 (__main__.TestModuleCUDA) ... miopen_batch_norm
miopen_batch_norm
miopen_batch_norm
miopen_batch_norm
ok
test_memory_format_nn_BatchNorm3d_cuda_float32 (__main__.TestModuleCUDA) ... miopen_batch_norm
miopen_batch_norm
miopen_batch_norm
miopen_batch_norm
ok

----------------------------------------------------------------------
Ran 2 tests in 3.516s

OK
```

Ignore the miopen_batch_norm prints it was a statement that I added when debuging.

**Review by jithunnair-amd — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
cc @sunway513

**Comment by jithunnair-amd** [comment]
pytorch-ci failed due to custom test artifacts logic not being applicable to older PyTorch commits: http://rocmhead.amd.com:8080/job/pytorch/job/pytorch-ci/105/
Updated custom test artifact zip/unzip logic to make it backward-compatible.

retest pytorch please

**Comment by jithunnair-amd** [comment]
retest pytorch please

**Comment by jithunnair-amd** [comment]
retest pytorch please

**Comment by jithunnair-amd** [comment]
@micmelesse Actually, this PR was incomplete I think, because it didn't include the changes to enable the two unit tests for ROCm: 
From http://rocmhead.amd.com:8080/job/pytorch/job/pytorch-test-2/69/consoleFull
```
05:11:37 test_memory_format_nn_BatchNorm2d_eval_mode_cuda_float32 (__main__.TestModuleCUDA) ... skipped "test doesn't currently work on the ROCm stack"
05:11:37 test_memory_format_nn_BatchNorm2d_eval_mode_cuda_float64 (__main__.TestModuleCUDA) ... skipped 'Skipped!'
05:11:37 test_memory_format_nn_BatchNorm2d_train_mode_cuda_float32 (__main__.TestModuleCUDA) ... skipped "test doesn't currently work on the ROCm stack"
05:11:37 test_memory_format_nn_BatchNorm2d_train_mode_cuda_float64 (__main__.TestModuleCUDA) ... skipped 'Skipped!'
05:11:37 test_memory_format_nn_BatchNorm3d_eval_mode_cuda_float32 (__main__.TestModuleCUDA) ... skipped "test doesn't currently work on the ROCm stack"
05:11:37 test_memory_format_nn_BatchNorm3d_eval_mode_cuda_float64 (__main__.TestModuleCUDA) ... skipped 'Skipped!'
05:11:37 test_memory_format_nn_BatchNorm3d_train_mode_cuda_float32 (__main__.TestModuleCUDA) ... skipped "test doesn't currently work on the ROCm stack"
05:11:37 test_memory_format_nn_BatchNorm3d_train_mode_cuda_float64 (__main__.TestModuleCUDA) ... skipped 'Skipped!'
```

Can you please file another PR to enable these unit tests on rocm5.3_internal_testing? We can then file an upstream PR that includes the changes in this and the other PR.

**Comment by micmelesse** [comment]
@jithunnair-amd Here is the pr, https://github.com/ROCmSoftwarePlatform/pytorch/pull/1063
